src/lib/transition_frontier_controller_tests/test_transaction_status.ml

```diff
@@ -0,0 +1,113 @@
+open Core
+open Coda_base
+open Async
+open Pipe_lib
+
+module Stubs = Stubs.Make (struct
+  let max_length = 5
+end)
+
+open Stubs
+
+let%test_module "transaction_status" =
+  ( module struct
+    module Test = Transaction_status.Make (struct
+      include Transition_frontier_inputs
+      module Transition_frontier = Transition_frontier
+      module Transaction_pool = Transaction_pool
+    end)
+
+    let logger = Logger.create ()
+
+    let trust_system = Trust_system.null ()
+
+    let key_gen =
+      let open Quickcheck.Generator in
+      let open Quickcheck.Generator.Let_syntax in
+      let keypairs = List.map Genesis_ledger.accounts ~f:fst in
+      let%map random_key_opt = of_list keypairs in
+      ( Genesis_ledger.largest_account_keypair_exn ()
+      , Signature_lib.Keypair.of_private_key_exn
+          (Option.value_exn random_key_opt) )
+
+    let create_pool ~frontier_broadcast_pipe =
+      let incoming_diffs, _ = Linear_pipe.create () in
+      let transaction_pool =
+        Transaction_pool.create ~logger ~trust_system ~incoming_diffs
+          ~frontier_broadcast_pipe
+      in
+      don't_wait_for
+      @@ Linear_pipe.iter (Transaction_pool.broadcasts transaction_pool)
+           ~f:(fun transactions ->
+             Logger.trace logger
+               "Transactions have been applied successfully and is propagated \
+                throughout the 'network'"
+               ~module_:__MODULE__ ~location:__LOC__
+               ~metadata:
+                 [ ( "transactions"
+                   , Transaction_pool.Resource_pool.Diff.to_yojson transactions
+                   ) ] ;
+             Deferred.unit ) ;
+      (* Need to wait for transaction_pool to see the transition_frontier *)
+      let%map () = after (Time.Span.of_sec 1.0) in
+      transaction_pool
+
+    let%test_unit "A pending transaction is either in the transition frontier \
+                   or transaction pool, but not in the best path of the \
+                   transition frontier" =
+      Async.Thread_safe.block_on_async_exn
+      @@ fun () ->
+      let%bind frontier =
+        create_root_frontier ~logger Genesis_ledger.accounts
+      in
+      let frontier_broadcast_pipe, _ = Broadcast_pipe.create (Some frontier) in
+      let%bind transaction_pool = create_pool ~frontier_broadcast_pipe in
+      let user_command =
+        Quickcheck.random_value
```

Comment:
I think @enolan was saying that `random_value` is actually deterministic -- we'd have to use `Quickcheck.test` to get actual random values. I think here the randomness isn't needed (as this isn't really a property based test), so maybe it's okay to leave this as is here.